test/integration/client/metrics/metrics_test.go

```diff
@@ -47,12 +46,12 @@ func TestAPIServerTransportMetrics(t *testing.T) {
 	featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllAlpha", true)
 	featuregatetesting.SetFeatureGateDuringTest(t, feature.DefaultFeatureGate, "AllBeta", true)
 
-	// reset default registry metrics
-	legacyregistry.Reset()
-
 	flags := framework.DefaultTestServerFlags()
 	flags = append(flags, "--runtime-config=api/all=true,api/beta=true")
-	result := kubeapiservertesting.StartTestServerOrDie(t, nil, flags, framework.SharedEtcd())
+	options := kubeapiservertesting.NewDefaultTestServerOptions()
```

Comment:
what is the benefit of moving it into the testserver? 
metrics are painful because are global, but hiding it into the testing type will make it failures still more surprising

Reply:
commented in the issue https://github.com/kubernetes/kubernetes/issues/128631#issuecomment-2847166013

/hold

hold for further discussion, apologize for the inconventience, the disagreement is not in the PR , is in the direction